### ISSUE TITLE
【feature request】

### ISSUE THREAD

**Issue by 45thhokage** [receyuki/stable-diffusion-prompt-reader#10]
A way to link this to your stable diffusion output folder with a gallery view

**Comment by KBLLR** [comment]
![Screenshot 2023-05-27 at 09 30 06](https://github.com/receyuki/stable-diffusion-prompt-reader/assets/84152245/cc831529-cb78-4b34-81f8-ef234c6eb727)

On mac M1 os13.4 if it helps, but I never had this message before when opening an app

**Comment by CodeWithMa** [comment]
> ![Screenshot 2023-05-27 at 09 30 06](https://user-images.githubusercontent.com/84152245/241366403-cc831529-cb78-4b34-81f8-ef234c6eb727.png)
> 
> On mac M1 os13.4 if it helps, but I never had this message before when opening an app

See: https://github.com/receyuki/stable-diffusion-prompt-reader#sd-prompt-readerapp-is-damaged-and-cant-be-opend-you-should-move-it-to-the-trash

PS: create your own issue next time

**Comment by receyuki** [comment]
> A way to link this to your stable diffusion output folder with a gallery view

I had no plans to do a folder/gallery view, but it seems like many people need it. Are you happy with dragging the output folder into the window every time you open the app? Or maybe select from open history also works for you? Or do you prefer to save the output path in the app? For now I recommend you using [diffusion toolkit](https://github.com/RupertAvery/DiffusionToolkit) instead.

**Comment by dsully** [comment]
+1 for a folder/gallery view. `diffusion toolkit` doesn't work for those of us who are Windows-free.

**Comment by 45thhokage** [comment]
> > A way to link this to your stable diffusion output folder with a gallery view
> 
> I had no plans to make a folder/gallery view before, but it seems like many people need it. Are you happy with dragging the output folder into the window every time you open the app? Or maybe select from open history also works for you? Or do you prefer to save the output path in the app? For now I recommend you using [diffusion toolkit](https://github.com/RupertAvery/DiffusionToolkit) instead.

I would prefer to have a saved output path set within the app. That being said, I would accept any of the alternative solutions. I've actually tried out diffusion toolkit before, and gotta say, the user interface isn't the greatest, but hey, it gets the job done and works just fine.

**Comment by receyuki** [comment]
Here is the story, over the past month I have tested the feasibility and performance of implementing gallery view in the current GUI framework(tkinter). It takes some time to read 4k images but it works. 

My original plan was to refactor the whole GUI with QT, so I contacted authors of some related libraries but still failed to solve some serious issues. So I decided to stick with the current GUI framework.

Anyway, here is my current design of the gallery view mode, any suggestions? BTW, user configuration including saved output path will be added as well.

<img width="1280" alt="Frame 4" src="https://github.com/receyuki/stable-diffusion-prompt-reader/assets/28808141/26524e9f-ec5a-417d-8ebd-6f1bda4ab9f2">

**Comment by 45thhokage** [comment]
That sounds and looks great. I'd like to suggest adding the capability to zoom into images. while pressing left you're able to change the lense of the zoom. other than that it looks great honestly

**Comment by receyuki** [comment]
> That sounds and looks great. I'd like to suggest adding the capability to zoom into images. while pressing left you're able to change the lense of the zoom. other than that it looks great honestly

Like the current single image gui?  Or open this image in anther window?

**Comment by 45thhokage** [comment]
> > That sounds and looks great. I'd like to suggest adding the capability to zoom into images. while pressing left you're able to change the lense of the zoom. other than that it looks great honestly
> 
> Like the current single image gui? Or open this image in anther window?

Yeah, on the single image gui somewhat similar to the vencord plugin for discord called "imageZoom"
